src/NuGet.Services.KillSwitch/HttpKillswitchService.cs

```
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace NuGet.Services.KillSwitch
+{
+    /// <summary>
+    /// A killswitch service that queries an HTTP service periodically for the list of activate killswitches.
+    /// </summary>
+    public class HttpKillswitchService : IKillswitchService
+    {
+        private readonly HttpClient _httpClient;
+        private readonly HttpKillswitchConfig _config;
+        private readonly ILogger<HttpKillswitchService> _logger;
+
+        private HashSet<string> _active;
+
+        public HttpKillswitchService(
+            HttpClient httpClient,
+            HttpKillswitchConfig config,
+            ILogger<HttpKillswitchService> logger)
+        {
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _config = config ?? throw new ArgumentNullException(nameof(config));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            _active = new HashSet<string>();
+        }
+
+        public void Start(CancellationToken cancellationToken)
+        {
+            Task.Run(async () =>
+            {
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    try
+                    {
+                        await RefreshAsync();
+                        await Task.Delay(_config.RefreshInterval);
+                    }
+                    catch (Exception e)
```

You are swallowing the exception and hiding issues. The correct behavior will be to add retries to protect against network issues, but throw on deserialization issues and other unrecoverable exceptions.

---

This runs on a background thread, throwing would cause the background thread to end.

That being said, I agree that we should have some "hard fail" behavior when we repeatedly fail to reload the killswitches. I'll make `IsActive` throw if the cache's staleness reaches some threshold.

---

Made `IsActive` throw if the cache reaches the configured staleness.